cookbook/security/voters.rst

```diff
@@ -1,205 +1,269 @@
 .. index::
-   single: Security; Voters
+   single: Security; Data Permission
+   single: Security: Voters
 
-How to Implement your own Voter to Blacklist IP Addresses
-=========================================================
+How to Use Voters to Check User Permissions
+===========================================
 
-The Symfony Security component provides several layers to authorize users.
-One of the layers is called a "voter". A voter is a dedicated class that checks
-if the user has the rights to connect to the application or access a specific
-resource/URL. For instance, Symfony provides a layer that checks if the user
-is fully authorized or if it has some expected roles.
+In Symfony, you can check the permission to access data by using the
+:doc:`ACL module </cookbook/security/acl>`, which is a bit overwhelming
+for many applications. A much easier solution is to work with custom voters,
+which are like simple conditional statements.
 
-It is sometimes useful to create a custom voter to handle a specific case not
-handled by the framework. In this section, you'll learn how to create a voter
-that will allow you to blacklist users by their IP.
+.. tip::
+
+    Take a look at the
+    :doc:`authorization </components/security/authorization>`
+    chapter for an even deeper understanding on voters.
+
+How Symfony Uses Voters
+-----------------------
+
+In order to use voters, you have to understand how Symfony works with them.
+All voters are called each time you use the ``isGranted()`` method on Symfony's
+security context (i.e. the ``security.context`` service). Each one decides
+if the current user should have access to some resource.
+
+Ultimately, Symfony uses one of three different approaches on what to do
+with the feedback from all voters: affirmative, consensus and unanimous.
+
+For more information take a look at
+:ref:`the section about access decision managers <components-security-access-decision-manager>`.
 
 The Voter Interface
 -------------------
 
 A custom voter must implement
 :class:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\VoterInterface`,
-which requires the following three methods:
+which has this structure::
+
+    interface VoterInterface
+    {
+        public function supportsAttribute($attribute);
+        public function supportsClass($class);
+        public function vote(TokenInterface $token, $object, array $attributes);
+    }
 
-.. include:: /cookbook/security/voter_interface.rst.inc
+The :method:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\VoterInterface::supportsAttribute`
+method is used to check if the voter supports the given user attribute (i.e:
+a role like ``ROLE_USER``, an ACL ``EDIT``, etc.).
```

Comment:
Complicate things -> having 2 items in a list of examples? Not really sure how this can complicate anything, it just explains some example attributes.